aptos-move/aptos-vm/src/verifier/resource_groups.rs

```diff
@@ -78,37 +91,39 @@ pub(crate) fn validate_resource_groups(
 /// * Verify all changes are compatible upgrades
 /// * Return any new members to validate correctness and all groups to assist in validation
 pub(crate) fn validate_module_and_extract_new_entries(
-    module_storage: &impl AptosModuleStorage,
-    module: &CompiledModule,
-    safer_resource_groups: bool,
+    module_storage: &impl ModuleStorage,
+    new_module: &CompiledModule,
+    features: &Features,
 ) -> VMResult<(
     BTreeMap<String, ResourceGroupScope>,
     BTreeMap<String, StructTag>,
 )> {
     let (new_groups, mut new_members) =
-        if let Some(metadata) = get_metadata_from_compiled_code(module) {
+        if let Some(metadata) = get_metadata_from_compiled_code(new_module) {
             extract_resource_group_metadata(&metadata)?
         } else {
             (BTreeMap::new(), BTreeMap::new())
         };
 
+    let old_module_if_exists =
+        module_storage.fetch_deserialized_module(new_module.address(), new_module.name())?;
     let (original_groups, original_members, mut structs) =
-        extract_resource_group_metadata_from_module(module_storage, &module.self_id())?;
+        extract_resource_group_metadata_from_module(old_module_if_exists)?;
 
-    for (member, value) in original_members {
+    for (member, group) in original_members {
         // We don't need to re-validate new_members above.
-        if Some(&value) != new_members.remove(&member).as_ref() {
+        if Some(&group) != new_members.remove(&member).as_ref() {
```

Comment:
does this need as_ref()?

Reply:
No if we change `&group` to `group`. Did that and removed `as_ref`

Reply:
gelash linter